deployment/helm/README.md

```diff
@@ -0,0 +1,166 @@
+
+4、edgex-kuiper's /kuiper/data directory is mapped to the host's /mnt/edgex-kuiper-data directory.
+
+- NodePort is enabled by default. According to default NodePort range(30000～32767), EdgeX NodePort mappings are as follows. 
+
+| EdgeX Service Name          | Exposed Port 
+| :-------------------------- | ------------- 
+| edgex-core-data             | 59880         
+| edgex-core-metadata         | 59881         
+| edgex-core-command          | 59882         
+| edgex-support-notifications | 59860         
+| edgex-support-scheduler     | 59861         
+| edgex-app-rules-engine      | 59701         
+| edgex-kuiper                | 59720         
+| edgex-device-rest           | 59986         
+| edgex-device-virtual        | 59900         
+| edgex-ui                    | 4000          
+| edgex-sys-mgmt-agent        | 58890         
+| edgex-redis                 | Not Exposed 
+| edgex-core-consul           | 8500          
+
+## Enabling security features
+
+The helm chart uses an Kubernetes ingress controller in lieu of a Kong API gateway.
+The ingress routes are configured to require client-side TLS authentication,
+which replaces the Kong JWT authentication method.
+
+### Prerequisites
+
+Before starting, make sure you have curl and openssl installed locally.
+These tools are need to generate TLS assets and test the configuration.
+
+These instructions also assume that your Kubernetes cluster has an installed
+ingress controller and an installed load balancer.
+The examples were built with a stock Ngnix ingress controller,
+and a MetalLB load balancer (with an assigned IP of 192.168.122.200).
+Adaptations may be needed if your cluster uses something different.
+
+To install the Nginx ingress controller,
+follow the instructions at
+<https://kubernetes.github.io/ingress-nginx/deploy/#quick-start>
+
+To install MetalLB,
+follow the instructions at
+<https://metallb.universe.tf/installation/#installation-with-helm>
+
+The example also arbitrarily configures the ingress route to
+respond to the hostname `edgex` 
+(which should be passed using the TLS Server Name Identification (SNI) feature).
+This can also be customized as well using the `edgex.security.tlsHost` setting.
+
+### Installation with security features
+
+Run the following two scripts to generate key material and install it into the cluster.
+
+The scripts create a server-side CA so that curl can trust the server,
+a client-side CA so that Nginx can trust the client,
+a server-side TLS certificate for the Nginx server to present to the client,
+and a client-side TLS certificate for the curl client to present to the server.
+
+```sh
+./tlsgen.sh
+./tlsinstall.sh
+```
+
+To install with security features enabled,
+set the `edgex.security.enabled` flag to `true` during installation.
+
+If necessary, uninstall the non-security one first.
+The helm chart is not coded to allow for dynamic switching in and out of secure mode.
```

Comment:
so you need to do a helm uninstall and then re-install?

Reply:
Yes.  Uninstall then reinstall.  I haven't tested anything different.  I make no claims that his helm cart is production-worthy.  The regression test suite hasn't even been run against it.